perl/lib/Sanger/CGP/Ascat/Implement.pm

```diff
@@ -414,8 +415,14 @@ sub sanitised_sample_from_bam {
 
 sub prepare {
   my $options = shift;
-  $options->{'tumour_name'} = sanitised_sample_from_bam($options->{'tumour'});
-  $options->{'normal_name'} = sanitised_sample_from_bam($options->{'normal'});
+  if ( $options->{'counts_input'} == 0 ) {
+    $options->{'tumour_name'} = sanitised_sample_from_bam($options->{'tumour'});
+    $options->{'normal_name'} = sanitised_sample_from_bam($options->{'normal'});
+  }
+  else {
+    $options->{'tumour_name'} = (split /\./ ,basename( $options->{'tumour'} ) )[0];
```

Comment:
I would prefer the user is required to provide these as arguments as there is no promise that the files will be named as we expect (in canpipe files are downloaded with just an ID, specifying the sample names is easier to work with).

Reply:
Basically once you determine that the input is counts files, then check for options args for t/n sample names.